### DISCUSSION TITLE
Please add support for INSERT RETURNING for PostgreSQL

### DISCUSSION THREAD

**Discussion by SpruceCloud** [rbatis/rbatis#292] [General]
Please add support for `INSERT RETURNING` in PostgreSQL database.

https://www.postgresql.org/docs/9.5/sql-insert.html

It can be
- either for returning new instance of struct filled with fields from DB
- or adding support for `last_insert_id` through it (there seems to be a discussion about this field in #75)

Example:

```
INSERT INTO distributors (did, dname) VALUES (DEFAULT, 'XYZ Widgets')
   RETURNING did;
```

**Comment by zhuxiujia** [comment]
I guess SQLX-core is not yet supported of INSERT RETURNING。
The current solution is to use predefined IDs, such as the snowflake algorithm and object-id. 2 is in the RBatis plugin

**Comment by zhuxiujia** [comment]
> Please add support for `INSERT RETURNING` in PostgreSQL database.
> 
> https://www.postgresql.org/docs/9.5/sql-insert.html
> 
> It can be
> 
> * either for returning new instance of struct filled with fields from DB
> * or adding support for `last_insert_id` through it (there seems to be a discussion about this field in [last_insert_id is None #75](https://github.com/rbatis/rbatis/issues/75))
> 
> Example:
> 
> ```
> INSERT INTO distributors (did, dname) VALUES (DEFAULT, 'XYZ Widgets')
>    RETURNING did;
> ```

 see https://github.com/rbatis/rbatis/blob/master/example/src/crud.rs

save_by_wrapper

**Comment by zhuxiujia** [comment]
v4 use impl_select! macro supported
